src/libraries/Microsoft.Extensions.FileProviders.Physical/src/PhysicalFilesWatcher.cs

```diff
@@ -487,7 +493,7 @@ internal static void RaiseChangeEvents(object state)
             public ChangeTokenInfo(
                 CancellationTokenSource tokenSource,
                 CancellationChangeToken changeToken,
-                Matcher matcher)
+                Matcher? matcher)
```

Comment:
is the previous constructor used anywhere? if not consider making Matcher non-null

Reply:
It's used on line 184

```cs
tokenInfo = new ChangeTokenInfo(cancellationTokenSource, cancellationChangeToken);
```